File: android/uhabits-android/src/main/java/org/isoron/uhabits/utils/SystemUtils.java
package org.isoron.uhabits.utils;

import android.app.*;
import android.content.*;
import android.os.*;
import android.view.*;


public class SystemUtils
{
    public static boolean isAndroidOOrLater()
    {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O;
    }

    public static void unlockScreen(Activity activity)
    {
        if (isAndroidOOrLater()) {
            KeyguardManager km =
                    (KeyguardManager) activity.getSystemService(Context.KEYGUARD_SERVICE);
            km.requestDismissKeyguard(activity, null);
        } else {
            activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD);
        }
    }
}

File: uhabits-android/src/main/java/org/isoron/uhabits/utils/SystemUtils.kt
/*
 * Copyright (C) 2016-2021 Álinson Santos Xavier <git@axavier.org>
 *
 * This file is part of Loop Habit Tracker.
 *
 * Loop Habit Tracker is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by the
 * Free Software Foundation, either version 3 of the License, or (at your
 * option) any later version.
 *
 * Loop Habit Tracker is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.isoron.uhabits.utils

import android.app.Activity
import android.app.KeyguardManager
import android.content.Context
import android.os.Build
import android.os.Build.VERSION.SDK_INT
import android.view.WindowManager

object SystemUtils {

    fun unlockScreen(activity: Activity) {
        if (SDK_INT >= Build.VERSION_CODES.O) {
            val km = activity.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
            km.requestDismissKeyguard(activity, null)
        } else {
            activity.window.addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD)
        }
    }
}
